### ISSUE TITLE
reflifc default constructibility of types

### ISSUE THREAD

**Issue by FireFlyForLife** [AndreyG/ifc-reader#36]
In principle, reflifc wraps an `AbstractReference` with the accompanying ifc `File`. These both have sane defaults. An `AbstractReference` has `AbstractReference::is_null()` and the file can be `nullptr`.

I wanted to start a discussion, if making these types default constructible was something you had already considered. I've found it nice if I can default construct them.

**Comment by AndreyG** [comment]
Yes, it could be nice to allow default constructibility of `reflifc` types. But the downside is that in such case these types should provide method `bool is_valid()` and every client must invoke either `if (!param.is_valid()) ...` or `assert(param.is_valid());`. Every method returning `reflifc::SomeType` must be annotated if it could return invalid value.  Every method accepting param of `reflifc::SomeType` should be documented if it's ready to handle invalid value.

I'm not sure if it's worth it. Could you, please, show the case when default construction of the `reflifc` type is needed?
